aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -106,6 +108,44 @@ Tensor isinf(const Tensor &self) {
   });
 }
 
+static inline TensorIterator build_is_infinity_op_iterator(Tensor& result, const Tensor& self) {
+  return at::TensorIteratorConfig()
+    .check_all_same_dtype(false)
+    .set_check_mem_overlap(true)
+    .add_output(result)
+    .add_input(self)
+    .cast_common_dtype_to_outputs(true)
+    .build();
+}
+
+Tensor isposinf(const Tensor &self) {
+  Tensor result = at::empty_like(self, at::kBool, at::MemoryFormat::Preserve);
+  at::isposinf_out(result, self);
+  return result;
+}
+
+Tensor& isposinf_out(Tensor& result, const Tensor& self) {
+  TORCH_CHECK(!self.is_complex(), "isposinf does not support complex inputs.");
```

Comment:
Better add a check for result being a bool tensor, too

Reply:
NumPy supports passing `non-bool` array to `out=` parameters, we may not add the check due to consistency.

Reply:
It's OK to be inconsistent as long we're MORE restrictive, because there are many NumPy features we haven't implemented yet. For example, we don't support taking the sin of integer types, unlike NumPy, and most of our unary functions require the output type be the same as the input type.

So it's OK to restrict the output to be bool here, if you like. It's also OK to let it be other types but a test will need to be added verifying it works properly. Either way is great.

Reply:
Thanks for the detail explanations, bool restricts for output are added as well as the test codes.